evalml/tests/automl_tests/test_automl.py

```diff
@@ -1058,32 +1058,40 @@ def test_catch_keyboard_interrupt(mock_fit, mock_score, mock_input,
 
 
 @patch('evalml.automl.automl_algorithm.IterativeAlgorithm.next_batch')
-@patch('evalml.automl.AutoMLSearch._evaluate')
-def test_pipelines_in_batch_return_nan(mock_evaluate, mock_next_batch, X_y_binary, dummy_binary_pipeline_class):
+@patch('evalml.automl.AutoMLSearch._evaluate_pipelines')
+def test_pipelines_in_batch_return_nan(mock_evaluate_pipelines, mock_next_batch, X_y_binary, dummy_binary_pipeline_class):
     X, y = X_y_binary
-    mock_evaluate.side_effect = [{'cv_score_mean': 0}, {'cv_score_mean': np.nan}]
+    mock_evaluate_pipelines.return_value = [0, np.nan]
     mock_next_batch.side_effect = [[dummy_binary_pipeline_class(parameters={}), dummy_binary_pipeline_class(parameters={})]]
     automl = AutoMLSearch(problem_type='binary', allowed_pipelines=[dummy_binary_pipeline_class])
-    automl.search(X, y)
+    # Mock rankings so `best_pipeline` setting does not error out
+    with patch('evalml.automl.AutoMLSearch.rankings', new_callable=PropertyMock) as mock_rankings:
```

Comment:
Why did you change `mock_evaluate_pipelines` from `.side_effect` to `.return_value`?

So we need this this new PropertyMock because you're mocking `_evaluate_pipelines` which is what calls `_add_result` (which is what populates the data for the ranking table)? 

I wonder if we even need this code block in the test. I think the point of the test is to verify that the search ends when it encounters a batch of all nan (which is what happens in the code block that doesn't have the property mock). 

This would be fine to merge as is but I'm guessing we can simplify this test a bit without losing any coverage.

Reply:
> Why did you change mock_evaluate_pipelines from .side_effect to .return_value?

Good question. I changed it to `.return_value` as I needed the mock `_evaluate_pipelines` to return the entire list of pipeline scores. Passing in a list to `.side_effect` cause it to iterate through the list and only return each element. 

RE: the code block, you're right about the reason why we have to populate the ranking table manually. My impression of the intent of this section was to show that having a `np.nan` score as one of the pipeline results won't result in raising the AutoMLSearchException. If this isn't a necessary check, then I'm good with cutting out this section.

Reply:
Maybe we can delete that code block then? The second code block returns a Nan in the second batch and the search doesn't terminate then but we can check that explicitly with `assert mock_evaluate_pipelines.call_count == 3`.

This isn't blocking merge but would be nice to remove code that provides redundant coverage.